### DISCUSSION TITLE
[Simulation question]:

### DISCUSSION THREAD

**Discussion by datboigbenga** [topas-nbio/TOPAS-nBio#51] [Simulation-question]
### Simulation question

Trying to use gamma source on Circular Plasmid example, DNAScorerStepByStep.txt .
But Scoring isn't giving results

### TOPAS-nBio category

Scoring

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Attestation

- [X] I testify that I have fully read the bug reporting guideline beforehand.

**Comment by tiagoaaazevedo1** [comment]
Hi! Could you give some more details about your setup?

One of the things to keep in mind is that photons interact very little with the environment, so it could be a low number of histories. 

Two things to kinda troubleshoot:
1. What is the dose that you are giving to the circular plasmid? 
2. If you visualize setup, are the photons reaching the geometry? 

Hope this helps! If you need further assistance, please don't hesitate to reach out.

Best regards,
Tiago Azevedo

**Reply by datboigbenga** [reply]
Thanks for the reply.
Please check out the code below

# TOPAS-nBio example : Circular DNA Plasmid
# DNA plasmid is placed in a box and irradiated by protons.
# ntuple scorer is used to score energy deposition in the DNA segment
# Geant4-DNA processes are only activated within the DNA plasmid ring


b:Ge/QuitIfOverlapDetected="true"

#World volume
d:Ge/World/HLX=150 nm
d:Ge/World/HLY=150 nm
d:Ge/World/HLZ=150 nm
b:Ge/World/Invisible="true"
s:Ge/World/Material="G4_WATER"

#Circular DNA plasmid
s:Ge/MyDNA/Type="tscircularplasmid"
s:Ge/MyDNA/Parent="World"
s:Ge/MyDNA/Material="G4_WATER"

#Define the number of base pairs in the ring
i:Ge/MyDNA/NumberOfBasePairs=200

# Material for volumes in strand 1
s:Ma/G4_WATER_MODIFIED_1/CloneFromMaterial = "G4_WATER"
d:Ma/G4_WATER_MODIFIED_1/CloneWithDensity  = 1.0 g/cm3

# Auxiliar material for volumes in strand 2
s:Ma/G4_WATER_MODIFIED_2/CloneFromMaterial = "G4_WATER_MODIFIED_1"
d:Ma/G4_WATER_MODIFIED_2/CloneWithDensity  = 1.0 g/cm3

s:Ge/MyDNA/Base/Material="G4_WATER"
s:Ge/MyDNA/BasePair/Material="G4_WATER_MODIFIED_1"
s:Ge/MyDNA/BasePair/Color="yellow"
s:Ge/MyDNA/BasePair/DrawingStyle="solid"
s:Ge/MyDNA/Backbone1/Material="G4_WATER_MODIFIED_1"
s:Ge/MyDNA/Backbone1/Color="green"
s:Ge/MyDNA/Backbone1/DrawingStyle="solid"
s:Ge/MyDNA/Backbone2/Material="G4_WATER_MODIFIED_2"
s:Ge/MyDNA/Backbone2/Color="blue"
s:Ge/MyDNA/Backbone2/DrawingStyle="solid"

sv:Ph/Default/Modules = 1 "g4em-dna"

#Define a beam type
s:So/Disk/Type = "Beam" 
s:So/Disk/Component =  "BeamPosition" 
s:So/Disk/BeamParticle = "gamma" #gamma
d:So/Disk/BeamEnergy = 5 MeV 
s:So/Disk/BeamShape = "Rectangle"
s:So/Disk/BeamPositionDistribution = "Flat"
s:So/Disk/BeamAngularDistribution = "Gaussian"
d:So/Disk/BeamAngularCutoffX = 10 deg
d:So/Disk/BeamAngularCutoffY = 10 deg
s:So/Disk/BeamPositionCutoffShape = "Rectangle"
d:So/Disk/BeamPositionCutoffX = 15 nm
d:So/Disk/BeamPositionCutoffY = 15 nm
u:So/Disk/BeamEnergySpread = 0 
d:So/Disk/BeamAngularSpreadX = 0.01 deg
d:So/Disk/BeamAngularSpreadY = 0.01 deg
i:So/Disk/NumberOfHistoriesInRun=500

# -- Scorer --
s:Sc/DNADamageScorer/Quantity				= "DNADamageStepByStep"
includeFile									= DNAScorerStepByStep.txt
s:Sc/DNADamageScorer/Component 				= "MyDNA"
i:Sc/DNADamageScorer/NumberOfHistoriesInRun = So/Disk/NumberOfHistoriesInRun
i:Sc/DNADamageScorer/BasePairPositionAtGeometricHierarchy = 0
sv:Sc/DNADamageScorer/Strand1MaterialNames  = 1 "G4_WATER_MODIFIED_1"
sv:Sc/DNADamageScorer/Strand2MaterialNames  = 1 "G4_WATER_MODIFIED_2"
s:Sc/DNADamageScorer/PrimaryParticle		= So/Disk/BeamParticle
d:Sc/DNADamageScorer/MeanEnergy				= So/Disk/BeamEnergy MeV

#Visualization
s:Gr/ViewA/Type="OpenGL"
i:Gr/ViewA/WindowSizeX = 1500
i:Gr/ViewA/WindowSizeY = 1500
d:Gr/ViewA/Theta = 55 deg
d:Gr/ViewA/Phi = 20 deg
d:Gr/ViewA/PerspectiveAngle = 30 deg

Ts/PauseBeforeQuit = "false"
b:Ts/ShowCPUTime = "True"


-------------------------------------------------------------------------------------------------


code for DNAScorerStepByStep.txt


# Setting up the output
b:Sc/DNADamageScorer/PropagateToChildren						= "true"
s:Sc/DNADamageScorer/OutputFile									= "DNADamage"
s:Sc/DNADamageScorer/OutputType									= "ascii"
s:Sc/DNADamageScorer/IfOutputFileAlreadyExists					= "overwrite"

# Options for direct damage (recommended values for spherical DNA model)
# d:Sc/DNADamageScorer/DirectDamageThreshold						= 11.75 eV    # For half-cylindrical DNA model, the recommended value is 17.5 eV
b:Sc/DNADamageScorer/UseLinearProbabilityForDirectDamage		= "true"
d:Sc/DNADamageScorer/LowerLimitForLinearProbabilityFunction	= 5 eV
d:Sc/DNADamageScorer/UpperLimitForLinearProbabilityFunction	= 37.5 eV

# Options for quasi-direct damage
u:Sc/DNADamageScorer/ProbabilityOfChargeTransferFromHydrationShellToBackbone = 0.33333

# Options for indirect damage (recommended values for spherical DNA model)
b:Sc/DNADamageScorer/AlwaysScavengeSpeciesInDNAComponents		= "false"
u:Sc/DNADamageScorer/ProbabilityOfScavengingInBackbone			= 0.25		  # For half-cylindrical DNA model, the recommended value is 0.0585
u:Sc/DNADamageScorer/ProbabilityOfScavengingInBase				= 1.0
u:Sc/DNADamageScorer/ProbabilityOfIndirectDamageToBackbone		= 0.55		  # If always scavenging in DNA compoennts, the recommended value is 0.4
u:Sc/DNADamageScorer/ProbabilityOfIndirectDamageToBase			= 1.0		  # If always scavenging in DNA compoennts, the recommended value is 0.4
b:Sc/DNADamageScorer/ScavengeInHistones							= "true"

# Options for damage classification
i:Sc/DNADamageScorer/MaximumBasePairDistanceToConsiderDSB		= 10
b:Sc/DNADamageScorer/ExcludeShortFragments						= "false"
#i:Sc/DNADamageScorer/LowerThresholdForFragmentDetection		= 0
#i:Sc/DNADamageScorer/UpperThresholdForFragmentDetection		= 3E8

# Options for foci scoring
b:Sc/DNADamageScorer/ScoreNumberOfFoci							= "true"
dv:Sc/DNADamageScorer/FociSizes									= 2 150 500 nm # Maximum of 5 sizes
b:Sc/DNADamageScorer/Get3DFociImage								= "true"
b:Sc/DNADamageScorer/Get2DFociImages							= "true"
sv:Sc/DNADamageScorer/2DFociImagePlanes							= 3 "X" "Y" "Z"
s:Sc/DNADamageScorer/MicroscopePSFShape							= "Gaussian"
d:Sc/DNADamageScorer/MicroscopePSFWidth							= 400 nm
dv:Sc/DNADamageScorer/Foci2DImageResolutions					= 2 25 100 nm
d:Sc/DNADamageScorer/Foci3DImageResolution						= 400 nm

# Options for output extension
b:Sc/DNADamageScorer/WriteCSVOutputWithAllDamageSpecification	= "true"
b:Sc/DNADamageScorer/ScoreDirectDamage							= "true"
b:Sc/DNADamageScorer/ScoreIndirectDamage						= "true"
b:Sc/DNADamageScorer/ScoreQuasiDirectDamage						= "true"
b:Sc/DNADamageScorer/ScoreBaseDamages							= "true"
b:Sc/DNADamageScorer/ScoreBackboneDamages						= "true"
b:Sc/DNADamageScorer/BreakDownOutputPerDamageOrigin				= "true"

# Options for SDD output specification
d:Sc/DNADamageScorer/DosePerExposure							= 3 Gy
b:Sc/DNADamageScorer/IncludeDSBDamageSitesOnlyInSDD				= "false"
b:Sc/DNADamageScorer/WriteMinimalSDDOutput						= "false"

# Data for SDD header
s:Sc/DNADamageScorer/AuthorName							   		= "Name"
s:Sc/DNADamageScorer/SimulationDetails					   		= "Nucleus simulation" 
s:Sc/DNADamageScorer/SourceDetails					   	   		= "Monoenergetic source"
i:Sc/DNADamageScorer/SourceType					   		   		= 1
s:Sc/DNADamageScorer/EnergyDistribution					   		= "M, 0"
s:Sc/DNADamageScorer/IrradiationTarget					   		= "Nucleus"
s:Sc/DNADamageScorer/CellCycleStage					       		= "0"
s:Sc/DNADamageScorer/DNAStructure					           	= "0, 1"
i:Sc/DNADamageScorer/InVitroOrInVivo						   	= 0
s:Sc/DNADamageScorer/ProliferationStatus					   	= "1"
s:Sc/DNADamageScorer/Microenvironment					       	= "20, 0.01"
d:Sc/DNADamageScorer/Time					  				   	= 0 s
s:Sc/DNADamageScorer/AdditionalInfo					       		= ""

**Reply by tiagoaaazevedo1** [reply]
Hi! So if you try with protons, even for 500 histories, it returns a result. The problem is really that with photons you have to be patient because they interact very little. What I normally due is that I add a second scorer that is keeping track of the dose, and then I use the parameter: 

d:Sc/MyScorer/RepeatSequenceUntilSumGreaterThan = ValueofDose Gy

This way you can set the simulation based on a certain value. But keep in mind that for photons this can run for a very long time until the desired dose is achieved.

**Reply by datboigbenga** [reply]
i'm still seeing no difference
